app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt

```diff
@@ -67,31 +69,35 @@ class DroneConnectActivity : AppCompatActivity() {
      * Connect a simulation to the application using an ip address and a port
      */
     fun connectSimulatedDrone(@Suppress("UNUSED_PARAMETER") view: View) {
+
         val ip = findViewById<EditText>(R.id.text_IP_address).text.toString()
+
         if (verifyIp(ip)) {
             val port = findViewById<EditText>(R.id.text_port).text.toString()
             showWaiting()
-            droneService.setSimulation(ip, port)
 
+            //Launch this task asynchronously
             GlobalScope.launch {
-                async {
-                    checkIfDroneConnected(30)
-
-                    val mainHandler = Handler(applicationContext.mainLooper)
-
-                    val myRunnable = Runnable {
-                        if (droneService.isConnected()) {
-                            val intent = Intent(applicationContext, ConnectedDroneActivity::class.java)
-                            startActivity(intent)
-                        } else {
-                            droneService.disconnect()
-                            showConnectionOptions()
-                            ToastHandler.showToastAsync(applicationContext, R.string.ip_connection_timeout)
-                        }
-                    }
+                droneService.setSimulation(ip, port)
+            }
+
+            //Launch this task asynchronously
+            GlobalScope.launch {
+
+                checkIfDroneConnected(30)
```

Comment:
The value (30) shouldn't be hardcoded but rather put in a const variable

Reply:
You are right, I corrected it.